src/containers/FacilityForm.js

```diff
@@ -74,53 +76,62 @@ function FacilityForm({ backend, history }) {
-      disabled={!Object.keys(requiredFields).every((key) => !!fields[key])}
-      description="Enter some information about your facility."
-    >
-      <InputText
-        label="Name of the facility"
-        value={dropSiteFacilityName}
-        customOnChange={handleFieldChange('dropSiteFacilityName')}
-      />
-      <InputText
-        label="Zip code"
-        value={dropSiteZip}
-        customOnChange={handleFieldChange('dropSiteZip')}
-      />
-      <div css={{ display: 'flex', '> *': { width: '50%' } }}>
-        <InputText
-          label="City"
-          value={dropSiteCity}
-          customOnChange={handleFieldChange('dropSiteCity')}
-        />
-        <InputDropdown
-          placeholder="State"
-          value={dropSiteState}
-          options={states}
-          customOnChange={handleFieldChange('dropSiteState')}
-        />
-      </div>
-      <InputText
-        label="Full Street Address"
-        value={dropSiteAddress}
-        customOnChange={handleFieldChange('dropSiteAddress')}
-      />
-      <InputText
-        label="Website URL (optional)"
-        value={dropSiteUrl}
-        customOnChange={handleFieldChange('dropSiteUrl')}
-      />
-      <Note>
+  const formData = {
+    description: 'Enter some information about your facility.',
+    disabled: !Object.keys(requiredFields).every((key) => !!fields[key]),
+    onSubmit: handleSubmit,
+    title: 'Add a new facility',
+    formFields: [
+      {
+        customOnChange: handleFieldChange('dropSiteFacilityName'),
+        label: 'Name of the facility',
+        type: formFieldTypes.INPUT_TEXT,
+        value: dropSiteFacilityName,
+      },
+      {
+        customOnChange: handleFieldChange('dropSiteZip'),
+        label: 'Zip code',
+        type: formFieldTypes.INPUT_TEXT,
+        value: dropSiteZip,
+      },
+      {
+        customOnChange: handleFieldChange('dropSiteCity'),
+        isHalfWidth: true,
+        label: 'City',
+        type: formFieldTypes.INPUT_TEXT,
+        value: dropSiteCity,
+      },
+      {
+        customOnChange: handleFieldChange('dropSiteState'),
+        isHalfWidth: true,
+        label: 'State',
+        options: states,
+        placeholder: 'State',
+        type: formFieldTypes.INPUT_DROPDOWN,
+        value: dropSiteState,
+      },
+      {
+        customOnChange: handleFieldChange('dropSiteAddress'),
+        label: 'Full Street Address',
+        type: formFieldTypes.INPUT_TEXT,
+        value: dropSiteAddress,
+      },
+      {
+        customOnChange: handleFieldChange('dropSiteUrl'),
+        label: 'Website URL (optional)',
+        type: formFieldTypes.INPUT_TEXT,
+        value: dropSiteUrl,
+      },
+    ],
+    otherContent: [
+      <Note key="note">
         If you’re working from a temporary facility, email us at{' '}
         <Anchor href="mailto:help@help.supply">help@help.supply</Anchor>
-      </Note>
-    </Form>
-  );
+      </Note>,
+    ],
+  };
+
+  return CreateForm(formData);
```

Comment:
This file represents an example of how this would be used. We don't save anything from a lines-of-code standard, but this approach is much more data driven and should make the Need Services forms easier to manage.